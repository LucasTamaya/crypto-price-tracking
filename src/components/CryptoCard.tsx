import { ICrypto } from "../types/crypto";

const CryptoCard: React.FC<ICrypto> = ({
  id,
  name,
  symbol,
  image,
  current_price,
  market_cap_rank,
}) => {
  return (
    <div className="flex flex-row items-center justify-between border border-white rounded-xl py-5 px-3 cursor-pointer transition-transform hover:scale-105">
      <div className="flex flex-row items-center gap-x-3">
        <img src={image} className="w-8 h-8" alt={`${name}`} />
        <div className="flex flex-col justify-center">
          <h3 className="text-white font-bold">{name}</h3>
          <p className="text-gray-400 uppercase">{symbol}</p>
        </div>
      </div>
      <div className="flex flex-col justify-center items-end gap-y-1">
        <p className="text-white font-bold">${current_price}</p>
        <div className=" flexflex-row justify-center items-center bg-yellow rounded-lg px-2">
          <p className="text-gray-400 font-bold">#{market_cap_rank}</p>
        </div>
      </div>
    </div>
  );
};

export default CryptoCard;
