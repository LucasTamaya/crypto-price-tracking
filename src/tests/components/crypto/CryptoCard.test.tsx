import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import CryptoCard from "../../../components/crypto/CryptoCard";

const MockedComponent = () => {
  return (
    <Router>
      <CryptoCard
        id="bitcoin"
        name="Bitcoin"
        symbol="BTC"
        image="https://btc.jpg"
        current_price={15000}
        market_cap_rank={1}
        key="bitcoin"
      />
    </Router>
  );
};

describe("CryptoCard Component", () => {
  it("should renders the component", () => {
    render(<MockedComponent />);

    expect(screen.getByRole("link")).toBeInTheDocument();
    expect(screen.getByRole("img")).toBeInTheDocument();
    expect(
      screen.getByRole("heading", { name: "Bitcoin" })
    ).toBeInTheDocument();
    expect(screen.getByText("BTC")).toBeInTheDocument();
    expect(screen.getByText("$15000")).toBeInTheDocument();
    expect(screen.getByText("#1")).toBeInTheDocument();
  });

  it("should matches the snapshot", () => {
    const component = renderer.create(<MockedComponent />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
