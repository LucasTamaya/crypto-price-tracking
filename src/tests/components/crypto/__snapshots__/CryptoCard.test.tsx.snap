// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoCard Component should matches the snapshot 1`] = `
<a
  className="flex flex-row items-center justify-between border border-white rounded-xl py-5 px-3 cursor-pointer transition-transform hover:scale-105"
  href="/crypto/bitcoin"
  onClick={[Function]}
>
  <div
    className="flex flex-row items-center gap-x-3"
  >
    <img
      alt="Bitcoin"
      className="w-8 h-8"
      src="https://btc.jpg"
    />
    <div
      className="flex flex-col justify-center"
    >
      <h3
        className="text-white font-bold"
      >
        Bitcoin
      </h3>
      <p
        className="text-gray-400 uppercase"
      >
        BTC
      </p>
    </div>
  </div>
  <div
    className="flex flex-col justify-center items-end gap-y-1"
  >
    <p
      className="text-white font-bold"
    >
      $
      15000
    </p>
    <div
      className=" flexflex-row justify-center items-center bg-yellow rounded-lg px-2"
    >
      <p
        className="text-gray-400 font-bold"
      >
        #
        1
      </p>
    </div>
  </div>
</a>
`;
